### PR TITLE
Test/test against server

### PR DESCRIPTION
**Pull request by ayinloya**
When tests are run for the first time, the responses will be saved and used in subsequent runs.

Add gem vcr to save http responses for tests
Add timecop to make signature constant for certain tests
Add test for get_job_status in web_api
Add SMILE_PARTNER_ID and SMILE_API_KE as environment variables in tests

### CODE REVIEW

File: spec/smile-identity-core/id_api_spec.rb [hunk 1]
```diff
@@ -1,16 +1,16 @@
...
+  let(:connection) { SmileIdentityCore::IDApi.new(partner_id, api_key, sid_server) }
 
   let(:partner_params) do
     {
-      user_id: 'dmKaJazQCziLc6Tw9lwcgzLo',
-      job_id: 'DeXyJOGtaACFFfbZ2kxjuICE',
-      job_type: SmileIdentityCore::JOB_TYPE::BASIC_KYC
+      user_id: SecureRandom.uuid,
+      job_id: SecureRandom.uuid,
+      job_type: SmileIdentityCore::JOB_TYPE::ENHANCED_KYC
```
Comment (tamssokari) [inline spec/smile-identity-core/id_api_spec.rb]: this is because `BASIC_KYC` returns "Match / No Match / Partial Match"

correct?

Comment (ayinloya) [reply spec/smile-identity-core/id_api_spec.rb]: That is right

File: spec/smile-identity-core/web_api_spec.rb [hunk 2]
```diff
@@ -1,17 +1,16 @@
...
 
 RSpec.describe SmileIdentityCore::WebApi do
-  let(:partner_id) { '001' }
+  let(:partner_id) { ENV.fetch('SMILE_PARTNER_ID') }
   let(:default_callback) { 'www.default_callback.com' }
-  let(:api_key) { Base64.encode64(OpenSSL::PKey::RSA.new(1024).public_key.to_pem) }
-  let(:sid_server) { SmileIdentityCore::ENV::TEST }
-
+  let(:api_key) { ENV.fetch('SMILE_API_KEY') }
+  let(:sid_server) { ENV.fetch('SMILE_SERVER_ENVIRONMENT', SmileIdentityCore::ENV::TEST) }
```
Comment (tamssokari) [inline spec/smile-identity-core/web_api_spec.rb:7]: can we have a `.env.sample` file for the development flow here, with a call-out to it in the README.

**Review by tamssokari — CHANGES_REQUESTED** [review]
seeing as we're using env variables, I reckon we should use `.env` for local development

File: spec/smile-identity-core/web_api_spec.rb (hunk 2 above)
Comment (ayinloya) [reply spec/smile-identity-core/web_api_spec.rb:7]: Yes, noted

**Comment by mldangelo** [comment]
This is getting stale. What are we planning to do with this?

**Comment by ayinloya** [comment]
> This is getting stale. What are we planning to do with this?

A lot of things have changed, will add all the updates and try to get this go in by close of next week

**Comment by mldangelo** [comment]
@ayinloya this is getting very stale. Let's close it for now and plan out e2e tests across all of our SDKs.

**Comment by ayinloya** [comment]
@mldangelo Sure, we needed to create a dedicated partner for these tests or use partner 002 credentials to these test could pass
Will close this for now until further planning is done

**Pull request closed without merging**